feat: allow parameters to be passed to custom commands

jcdickinson opened the pull request:
This allows parameters to be passed to custom commands, for example:

```lua
		vim.g.codegpt_commands = {
			test = {
				user_message_template = "I have the following {{language}} code: ... {{command_args}}. {{language_instructions}} Only return the unit tests. Only return the code snippet and nothing else. ",
				callback_type = "code_popup",
                -- ...
			},
		}
```

This can be invoked with `:Chat test some additional information that can help ChatGPT write better tests.`

dpayne commented:
So this was on purpose. I ran into a couple of usages where the first work of an instruction would overlap with a command. For example, if there is an "optimize" command and someone gives the instructions "Optimize this code to use less memory".

Do you think you could a config option where you can toggle between these two behaviors?
```
     if 1 == #opts.fargs and CommandsList.get_cmd_opts(opts.fargs[1]) ~= nil then
     if 1 <= #opts.fargs and CommandsList.get_cmd_opts(opts.fargs[1]) ~= nil then
```

jcdickinson commented:
@dpayne sorry about taking a while to get back to this. I have made the change that you requested. With the changes if `{{command_args}}` is present in one of the templates then it will enable the new behavior, otherwise it will revert to the old behavior.

I also fixed a bug that I introduced (or that always existed?). `get_cmd_opts` would raise an error for not-found commands, causing any any prompt not beginning with a known command to fail (instead of falling back to `code_edit`, which seems to be the desired behavior). `run_cmd` is also checking for unknown commands, so the logic was duplicated.

I also updated the `error()` calls to `vim.notify`, as that tends to be less obnoxious than `error()` (depending on the plugin setup).

Finally, I documented the templates.

dpayne commented:
Looks good, the extra fixes and documentation are appreciated, thank you!
